### PR TITLE
vk: remove unnecessary flush and gc

### PR DESCRIPTION
**Pull request by poweifeng**

### CODE REVIEW

File: filament/backend/src/vulkan/VulkanDriver.cpp [hunk 1]
```diff
@@ -1370,10 +1372,6 @@ void VulkanDriver::commit(Handle<HwSwapChain> sch) {
 
     VulkanSwapChain* swapChain = mResourceAllocator.handle_cast<VulkanSwapChain*>(sch);
 
-    if (mCommands->flush()) {
-        collectGarbage();
-    }
```
Comment (pixelflinger) [inline filament/backend/src/vulkan/VulkanDriver.cpp:1375]: wouldn't make more sense to flush here as opposed to in endFrame? e.g. on GL eglSwapBuffers() makes an implicit flush.

File: filament/backend/src/vulkan/VulkanSwapChain.cpp [hunk 2]
```diff
@@ -88,8 +88,8 @@ void VulkanSwapChain::present() {
                 .layerCount = 1,
         };
         mColors[mCurrentSwapIndex]->transitionLayout(cmdbuf, subresources, VulkanLayout::PRESENT);
-        mCommands->flush();
     }
+    mCommands->flush();
```
Comment (pixelflinger) [inline filament/backend/src/vulkan/VulkanSwapChain.cpp:92]: oh there is a flush in present. so do we need the one in endFrame()?

File: filament/backend/src/vulkan/VulkanDriver.cpp (hunk 1 above)
Comment (poweifeng) [reply filament/backend/src/vulkan/VulkanDriver.cpp:1375]: yes, it's been moved inside present

File: filament/backend/src/vulkan/VulkanSwapChain.cpp (hunk 2 above)
Comment (poweifeng) [reply filament/backend/src/vulkan/VulkanSwapChain.cpp:92]: for reasons I don't understand yet - Yes, we do need it in `endFrame()`.  I see an increase in `cpu` usage when it's not there, on both MoltenVK and adreno. I plan to turn on a performance validation on an nvidia card to see if that tells me anything.

**Review by pixelflinger — APPROVED** [review]

**Pull request merged**